x/programs/cmd/simulator/src/lib.rs

```diff
@@ -17,7 +17,7 @@ mod id;
 pub use id::Id;
 
 /// The endpoint to call for a [Step].
-#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[derive(Debug, Serialize, PartialEq, Clone)]
```

Comment:
I have removed `Deserialize` for types which only made sense implementing `Serialize`, should I back this off ?